app/assets/javascripts/components/course_creator/course_form.jsx

```diff
@@ -17,7 +17,11 @@ const CourseForm = (props) => {
     props.updateCourseProps({ private: isPrivate });
     props.updateCourseAction('private', isPrivate);
   };
-
+  const updateTaSupport = (e) => {
+    const ta_support = e.target.checked;
+    props.updateCourseProps({ ta_support: ta_support });
```

Comment:
I'm not really clear on why we have these two separate functions, updateCourseProps and updateCourseAction. It seems like some components use one, some use the other, and only one uses both. I suspect that the one using both is not a great pattern to follow. What is your understanding of these two functions and what they are for?

Also, the less login that lives in this file, the better; I think this new input should be extracted to its own component like some of the other ones in the CourseForm.

Reply:
```props.updateCourseAction('private', isPrivate);``` in ```course_form.jsx``` is a function passed down by ```course_creator.jsx``` which isn't actually updating the props on it's own. My guess is that the name has "action" in it to tell that it triggers an action to update the prop.

```updateCourseAction``` is a prop name given to  ```updateCourse``` function while passing it down from ```course_creator.jsx``` to ```course_form.jsx```

```
  updateCourse(key, value) {
    this.props.updateCourse({ [key]: value });
    if (includes(['title', 'school', 'term'], key)) {
      return this.props.setValid('exists');
    }
  },
```
Here you can see it basically just adds a check to call setValid and other than that it is just doing the same thing as ```props.updateCourseProps({ ta_support: ta_support });``` does, by converting the input values to a ```key: value``` pair.

As you can see in line **427 ./course_creator.jsx**. ```updateCourseProps``` is just a prop name given to ```this.props.updateCourse```

Reply:
Made another component ```course_checkbox.jsx``` and moved both `ta_support checkbox` and `privacy checkbox` to it

Reply:
So `updateCourseProps` is basically just a wrapper around `updateCourse` which also something related to validation for the title, school and term inputs, while `updateCourseAction` is just the plain `updateCourse` redux action. The title/school/term validation is not relevant for either the new checkbox or the existing one that you've refactored, so I think you can use only simpler one (`updateCourseAction`). Definitely don't use both.